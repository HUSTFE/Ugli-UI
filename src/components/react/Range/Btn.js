import React from 'react'
import PropTypes from 'prop-types'

const RangeBtn = ({ style = {}, onTouchStart, onTouchEnd, onTouchMove }) => {
  return (
    <div
      className="ugli-range-btn ugli-range-middle"
      style={style}
      onTouchStart={onTouchStart}
      onTouchEnd={onTouchEnd}
      onTouchMove={onTouchMove}
    />
  )
}

RangeBtn.propTypes = {
  onTouchStart: PropTypes.func.isRequired,
  onTouchEnd: PropTypes.func.isRequired,
  onTouchMove: PropTypes.func.isRequired,
}

export default RangeBtn
