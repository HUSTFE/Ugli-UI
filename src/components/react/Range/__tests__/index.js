import React from 'react'
import renderer from 'react-test-renderer'
import Btn from '../Btn'

describe('RangeButton', () => {
  it('RangeBtn render without crash', () => {
    const component = renderer.create(<Btn offset={2} />)
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })
})
