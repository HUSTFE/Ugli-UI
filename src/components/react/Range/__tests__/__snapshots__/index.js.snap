// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeButton RangeBtn render without crash 1`] = `
<div
  className="ugli-range-btn ugli-range-middle"
  style={Object {}}
/>
`;
